src/ace/managers/mouse.c

```diff
@@ -1,126 +1,128 @@
 #include <ace/managers/mouse.h>
+#include <ace/managers/log.h>
+#include <ace/utils/custom.h>
+#include <ace/generic/screen.h>
 
 /* Globals */
 tMouseManager g_sMouseManager;
 
 /* Functions */
-void mouseOpen() {
+void mouseCreate(UBYTE ubPortFlags) {
 #ifdef AMIGA
-	g_sMouseManager.pInputMP = CreatePort(NULL, 0L);
-	g_sMouseManager.pInputIO = (struct IOStdReq *) CreateExtIO(g_sMouseManager.pInputMP, sizeof(struct IOStdReq));
-
-	OpenDevice("input.device", 0, (struct IORequest *) g_sMouseManager.pInputIO, 0);
-#endif // AMIGA
-}
-
-void mouseSetState(UBYTE ubMouseCode, UBYTE ubMouseState) {
-#ifdef AMIGA
-	ubMouseCode -= IECODE_LBUTTON;
-#endif // AMIGA
-	g_sMouseManager.pStates[ubMouseCode] = ubMouseState;
-}
-
-UBYTE mouseCheck(UBYTE ubMouseCode) {
-#ifdef AMIGA
-	ubMouseCode -= IECODE_LBUTTON;
-#endif // AMIGA
-	return g_sMouseManager.pStates[ubMouseCode] != MOUSE_NACTIVE;
-}
-
-UBYTE mouseUse(UBYTE ubMouseCode) {
-#ifdef AMIGA
-	ubMouseCode -= IECODE_LBUTTON;
-#endif // AMIGA
-	if (g_sMouseManager.pStates[ubMouseCode] == MOUSE_ACTIVE) {
-		g_sMouseManager.pStates[ubMouseCode] = MOUSE_USED;
-		return 1;
+	mouseSetBounds(0, 0, SCREEN_PAL_WIDTH, SCREEN_PAL_HEIGHT);
+
+	g_sMouseManager.ubPortFlags = ubPortFlags;
+	g_sMouseManager.uwPrevPotGo = custom.potinp;
+	UWORD uwPotMask = 0;
+
+	// Enable RMB & MMB
+	if(ubPortFlags & MOUSE_PORT_1) {
```

Comment:
This IFs should be totally a function with few variables. Only 1 hardcoded element varies.

Reply:
Not really, since there is no need for that. Also, `mouseCreate()` is short enough to have this stuff written as it is right now. No need for moving every repetitive code to separate fn.